framework/src/parser/MooseSyntax.C

```diff
@@ -99,6 +99,7 @@ associateSyntax(Syntax & syntax, ActionFactory & action_factory)
 
   registerSyntax("AddOutputAction", "Outputs/*");
```

Comment:
It is certainly easy enough to do. However, there are several places where we have to fall back to what we have now because multiple types are associated with a particular syntax. I am not sure I like having two ways for doing this, although I agree it does make the single type association look a little nicer.